metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
Sorry, that may not have been real clear.  To complete the thought, 

method added to ParserBolt:

@Override
public void updateConfig(String path, byte[] data) throws IOException {
super.updateConfig(path, data);
if (path.startsWith(ConfigurationType.PARSER.getZookeeperRoot() + "/" + getSensorType())) {
atomicCheckAndSet(true);
}}

and in execute() method, just before invoking parser.parseOptional(), do:

if (atomicCheckAndSet(false)) { parser.configurationUpdated(getSensorParserConfig().getParserConfig());}